test/minishift/test_suite.go

```diff
@@ -0,0 +1,172 @@
+package minishift
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/fabric8-services/fabric8-tenant/auth"
+	"github.com/fabric8-services/fabric8-tenant/cluster"
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/fabric8-services/fabric8-tenant/environment"
+	"github.com/fabric8-services/fabric8-tenant/openshift"
+	"github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/gormsupport"
+	"github.com/fabric8-services/fabric8-tenant/test/resource"
+	"github.com/fabric8-services/fabric8-tenant/test/stub"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+// TestSuite is a base for tests using Minishift and gorm db
+type TestSuite struct {
+	gormsupport.DBTestSuite
+	ClusterService  *stub.ClusterService
+	AuthService     *stub.AuthService
+	Config          *configuration.Data
+	toReset         func()
+	minishiftConfig *Data
+}
+
+func (s *TestSuite) SetupTest() {
+	resource.Require(s.T(), resource.Database)
+
+	config, err := NewData()
+	require.NoError(s.T(), err)
+	if config.GetMinishiftURL() == "" {
+		s.T().Skipf("Minishift URL using a variable [%s] has to be set", varMinishiftURL)
```

Comment:
This looks confusing to me. because you have to set `MINISHIFT_URL` not `minishift.url`. maybe rewording is enough.(Do we want to set ENV or something else)

Reply:
well, this is getting to be complicated - go don't even have `MINISHIFT_URL` but `F8_MINISHIFT_URL`. I'll reword it not to use the variable name - I'll use:
`Test is skipped because of missing Minishift variables`

Reply:
sounds good enough

Reply:
changed in 91b9d52